packages/flutter/lib/src/material/flat_button.dart

```diff
@@ -37,14 +39,21 @@ import 'theme.dart';
 /// Flat buttons have a minimum size of 88.0 by 36.0 which can be overidden
 /// with [ButtonTheme].
 ///
+/// When descending from a [Theme] with an [AdaptiveWidgetThemeData] that
```

Comment:
This paragraph is difficult to parse.  I think a couple of sentences are warranted, beginning with something like: This is a platform-adaptive widget. When the target platform is iOS a flat style [Cupertino] button is created. On other platforms a Material Design Text Button (https://material.io/design/components/buttons.html) is created instead. The default platform-specific adaptions can be customized with [ThemeData.adaptiveWidgetTheme].